django_enumfield/enum.py

```diff
@@ -1,90 +1,109 @@
+from __future__ import absolute_import
+
 import logging
+from enum import Enum as NativeEnum, IntEnum as NativeIntEnum
 
 from django.utils import six
-from django.utils.encoding import python_2_unicode_compatible
+from django.utils.decorators import classproperty
 
 from django_enumfield.db.fields import EnumField
 
-
 logger = logging.getLogger(__name__)
+RAISE = object()
+
 
+class BlankEnum(NativeEnum):
+    BLANK = ""
 
-class EnumType(type):
-    """ Custom metaclass for Enum type """
+    @property
+    def label(self):
+        return ""
 
-    def __new__(mcs, *args):
-        """ Create enum values from all uppercase class attributes and store them in a dict on the Enum class."""
-        enum = super(EnumType, mcs).__new__(mcs, *args)
-        attributes = [k_v for k_v in list(enum.__dict__.items()) if k_v[0].isupper()]
-        labels = enum.__dict__.get('labels', {})
 
-        enum.values = {}
-        for attribute in attributes:
-            enum.values[attribute[1]] = enum.Value(attribute[0], attribute[1], labels.get(attribute[1]), enum)
-        return enum
+class ClassmethodShortcut(object):
+    def __init__(self, property_name, class_method):
+        self.property_name = property_name
+        self.class_method = class_method
 
+    def __get__(self, instance, owner):
+        if instance is None:
+            # class
+            return getattr(owner, self.class_method)
+        return getattr(instance, self.property_name)
 
-class Enum(six.with_metaclass(EnumType)):
+
+class Enum(NativeIntEnum):
     """ A container for holding and restoring enum values """
 
-    @python_2_unicode_compatible
-    class Value(object):
-        """
-        A value represents a key-value pair with a uppercase name and a integer value:
-        GENDER = 1
-        "name" is a upper case string representing the class attribute
-        "label" is a translatable human readable version of "name"
-        "enum_type" is the value defined for the class attribute
-        """
+    __labels__ = {}
+    __default__ = None
+    __transitions__ = {}
 
-        def __init__(self, name, value, label, enum_type):
-            self.name = name
-            self.value = value
-            self._label = label
-            self.enum_type = enum_type
+    # Hack to make both Enum.VALUE.name and Enum.name(Enum.VALUE) available
```

Comment:
This trick seems to confuse mypy. I'm getting following error when type-checking following code:
```
# [mypy] Argument 1 to "set_status" of "FeedImport" has incompatible type "int"; expected "ImportStatus" [Error]
feed_import.set_status(enums.ImportStatus.RUNNING)
```
Removing `name` and `label` solve the error.

Reply:
Looks like ignoring those two lines works as well.